### PR TITLE
Add WP_Site_Query support

### PR DESCRIPTION
**Pull request by JasonHoffmann**
Adds support for WP_Site_Query. Need to do a check before the process is even registered to ensure that we are in a multisite WordPress install. If we are not, the process will not register or be shown in the backend.

Unit Tests have been added to their own file. I’ll refactor everything to be like this eventually, and this let me tweak the requirements for the unit tests to only run when Multisite is enabled.

### CODE REVIEW

File: includes/functions.php [hunk 1]
```diff
@@ -31,6 +32,12 @@ function register_batch_process( $args ) {
 			$batch_process = new Users();
 			$batch_process->register( $args );
 			break;
+		case 'site':
+			if ( is_multisite() ) {
```
Comment (jjeaton) [inline includes/functions.php]: what happens here if we're not on multisite?

File: tests/types/test-batch-sites.php [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+		$annotations = $this->getAnnotations();
+
+		foreach ( array( 'class', 'method' ) as $depth ) {
+			if ( empty( $annotations[ $depth ]['requires'] ) ) {
+				continue;
+			}
+
+			$requires = array_flip( $annotations[ $depth ]['requires'] );
+			if ( isset( $requires['Multisite'] ) && ! is_multisite() ) {
+				$this->markTestSkipped( 'Multisite must be enabled' );
```
Comment (jjeaton) [inline tests/types/test-batch-sites.php]: 👍

**Comment by jjeaton** [comment]
```FILE: /home/scrutinizer/build/includes/batches/class-batch-sites.php
    ----------------------------------------------------------------------
    FOUND 1 ERROR AFFECTING 1 LINE
    ----------------------------------------------------------------------
     32 | ERROR | [x] Each line in an array declaration must end in a
        |       |     comma
    ----------------------------------------------------------------------
    PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
    ----------------------------------------------------------------------```

easy peasy

File: includes/functions.php (hunk 1 above)
Comment (JasonHoffmann) [inline includes/functions.php]: For now, if you are not on multisite, the process never even gets registered. So it won’t show up in the admin at all and can’t be run.

**Pull request merged**